### ISSUE TITLE
Misleading lex error when forgetting to access qualified identifier

### ISSUE THREAD

**Issue by colinwahl** [purescript/purescript#3801]
This is a small nit, but the lex error when forgetting to access a qualified identifier seems a bit misleading. I think we could probably make this particular error more specific for the actual cause, which (imo) is likely that the user forgot to finish what they were typing. (For example, today I was looking around my file for strange whitespace characters without realizing I had just forgotten to finish what I was typing!)

For example, in the following code:

```
bar :: Int
bar = Foo. 
```

You get the lex error
`Illegal whitespace character U+0020`

@natefaubion suggested we could have an error that says `Whitespace not allowed in qualified identifier` or similar.

You get a similar error for when you forget to access an unqualified identifier, such as

```
bar :: Int
bar = foo.
```

resulting in
`Unexpected or mismatched indentation`

(I'm not sure if the second one is as easy to clean up, but I think the first example is relatively straightforward)

Has anyone else had a similar experience, or have a suggestion for a more specific error message?

**Comment by hdgarrood** [comment]
"Whitespace not allowed in qualified identifer" sounds okay to me, although I wonder if something like "Unexpected whitespace, expected identifier" might be clearer?

**Comment by colinwahl** [comment]
I agree that "Unexpected whitespace, expected identifier" is clearer. Perhaps we should go with that

**Comment by igrep** [comment]
I got puzzled with that kind of error too.
I often make a mistake like below due to the habit of a Haskeller:

```purs
foreign import data Foo -- missing a kind signature!

foreign import otherFunc :: Foreign -> Foreign
```
